aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -128,6 +121,141 @@ Tensor _inverse_helper_cuda_lib(const Tensor& self) {
   return self_inv_working_copy;
 }
 
+template<typename scalar_t>
+inline static void _apply_svd_lib_gesvdj(const Tensor& self, Tensor& U, Tensor& S, Tensor& VT, Tensor& infos, bool compute_uv) {
+  using value_t = typename c10::scalar_value_type<scalar_t>::type;
+  auto self_data = self.data_ptr<scalar_t>();
+  auto U_data = U.data_ptr<scalar_t>();
+  auto S_data = S.data_ptr<value_t>();
+  auto VT_data = VT.data_ptr<scalar_t>();
+  auto self_stride = matrixStride(self);
+  auto U_stride = matrixStride(U);
+  auto S_stride = S.size(-1);
+  auto VT_stride = matrixStride(VT);
+  auto batchsize = batchCount(self);
+
+  int m = cuda_int_cast(self.size(-2), "m");
+  int n = cuda_int_cast(self.size(-1), "n");   
+
+  for(int i = 0; i < batchsize; i++){
```

Comment:
isn't batchsize an int64_t because it's returned by batchCount? Does line 135 need to call cuda_int_cast? 

Or, if this is guaranteed to always be called with an integer number of batches, it seems like this should check for that property, and we should ensure there's no a type mismatch between the loop variable and the object it's compared with.

Reply:
I think when working with Tensor dimensions it's best to default to int64_t unless there's a performance reason why not to.

Reply:
Added safe downcast for `batchsize` int64_t -> int